### PR TITLE
api: add call to list passwords

### PR DESCRIPTION
**Pull request by rithujohn191**
1.  Added a ListPasswords() method to the storage interface 
        1.1 updated all implementations of the interface.
2.  Added a gRPC call for listing passwords.

### CODE REVIEW

File: storage/conformance/conformance.go [hunk 1]
```diff
@@ -285,13 +285,30 @@ func testPasswordCRUD(t *testing.T, s storage.Storage) {
...
+	var passwordList []storage.Password
+	passwordList = append(passwordList, password)
+
+	listAndCompare := func(want []storage.Password) {
+		passwords, err := s.ListPasswords()
+		if err != nil {
+			t.Errorf("list password: %v", err)
+			return
+		}
+		if diff := pretty.Compare(want, passwords); diff != "" {
```
Comment (ericchiang) [inline storage/conformance/conformance.go]: We probably need to sort these before comparing them, right?

File: server/api.go [hunk 2]
```diff
@@ -171,3 +171,27 @@ func (d dexAPI) GetVersion(ctx context.Context, req *api.VersionReq) (*api.Versi
...
+	if err != nil {
+		log.Printf("api: failed to list passwords: %v", err)
+		return nil, fmt.Errorf("list passwords: %v", err)
+	}
+
+	var passwords []*api.Password
+	for _, password := range passwordList {
+		p := api.Password{
+			Email:    password.Email,
+			Hash:     password.Hash,
```
Comment (ericchiang) [inline server/api.go]: Can we not return the Hash? I don't think there's a good reason for us to do this.

**Comment by rithujohn191** [comment]
Closes #679.

@ericchiang

**Comment by ericchiang** [comment]
lgtm. Can you run the Kubernetes conformance tests and report back?

**Comment by rithujohn191** [comment]
PASS
ok  	github.com/coreos/dex/storage/kubernetes	1.584s

Kubernetes conformance tests passed

**Pull request merged**